sdk/eventhub/azure-eventhub/azure/eventhub/aio/_transport/_pyamqp_transport_async.py

```diff
@@ -262,7 +262,11 @@ async def _receive_task(consumer):
             while retried_times <= max_retries and running and consumer._callback_task_run:
                 try:
                     await consumer._open() # pylint: disable=protected-access
-                    running = await cast(ReceiveClientAsync, consumer._handler).do_work_async(batch=consumer._prefetch)
+                    if len(consumer._message_buffer) < max_batch_size:
```

Comment:
max_batch_size is hardcoded to 1 no matter what and we only read 1 message off the buffer at a time, so this will ensure we are only sending a flow frame of 300 every time we have processed ALL of the msgs out of the buffer

Reply:
might be worth a comment in the code